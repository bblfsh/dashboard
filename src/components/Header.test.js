import React from 'react';
import { shallow } from 'enzyme';
import Header, { DriverCode } from './Header';
import renderer from 'react-test-renderer';

const testLanguages = {
  auto: { name: 'auto' },
  python: { name: 'Python', url: 'Python.driver' },
  java: { name: 'Java', url: 'Java.driver' },
};

describe('DriverCode', () => {
  it('contains the driver URL if selectedLanguage is not auto', () => {
    const component = renderer.create(
      <DriverCode languages={testLanguages} actualLanguage='java' selectedLanguage='java' />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('contains the actualLanguage URL if is auto', () => {
    const component = renderer.create(
      <DriverCode languages={testLanguages} actualLanguage='python' selectedLanguage='auto' />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('Header', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Header 
        languages={testLanguages}
        actualLanguage='java'
        selectedLanguage='auto' />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('calls onLanguageChanged when the dropdown is changed', () => {
    const onLanguageChangedSpy = jest.fn();
    const wrapper = shallow(
      <Header 
        languages={testLanguages}
        onLanguageChanged={onLanguageChangedSpy}
        actualLanguage='java'
        selectedLanguage='auto' />
    );

    wrapper.find('#language-selector').simulate('change', {
      target: { value: 'python' },
    });
    expect(onLanguageChangedSpy.mock.calls.length).toBe(1);
    expect(onLanguageChangedSpy.mock.calls[0][0].target.value).toBe('python');
  });

  it('calls onRunParser when the button is clicked', () => {
    const onRunParserSpy = jest.fn();
    const wrapper = shallow(
      <Header 
        languages={testLanguages}
        onRunParser={onRunParserSpy}
        actualLanguage='java'
        selectedLanguage='auto' />
    );

    wrapper.find('#run-parser').simulate('click');
    expect(onRunParserSpy.mock.calls.length).toBe(1);
  });

  it('has the button disabled if is loading, even if user has typed', () => {
    const component = renderer.create(
      <Header 
        languages={testLanguages}
        loading={true}
        dirty={true}
        actualLanguage='python'
        selectedLanguage='auto' />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('has the button disabled if is not loading but user has not typed', () => {
    const component = renderer.create(
      <Header 
        languages={testLanguages}
        loading={false}
        dirty={false}
        actualLanguage='python'
        selectedLanguage='auto' />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('has the button enabled if user has typed and is not loading', () => {
    const component = renderer.create(
      <Header 
        languages={testLanguages}
        loading={false}
        dirty={true}
        actualLanguage='python'
        selectedLanguage='auto' />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
