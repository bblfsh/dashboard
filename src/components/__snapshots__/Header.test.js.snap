// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriverCode contains the actualLanguage URL if is auto 1`] = `
.lhXvrk {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: .5rem;
}

.jUlwLK {
  width: 20px;
  opacity: .7;
  margin-right: .3rem;
}

.bnLiGR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: .9rem;
}

.bnLiGR:hover {
  color: #EB7422;
}

.bnLiGR:hover .sc-VigVT {
  opacity: 1;
}

<div
  className="sc-gqjmRU lhXvrk"
>
  <a
    className="sc-jTzLTM bnLiGR"
    href="Python.driver"
    target="_blank"
  >
    <img
      alt="Driver code repository on GitHub"
      className="sc-VigVT jUlwLK"
      src="github.svg"
      title="Driver code repository on GitHub"
    />
  </a>
</div>
`;

exports[`DriverCode contains the driver URL if selectedLanguage is not auto 1`] = `
.lhXvrk {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: .5rem;
}

.jUlwLK {
  width: 20px;
  opacity: .7;
  margin-right: .3rem;
}

.bnLiGR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: .9rem;
}

.bnLiGR:hover {
  color: #EB7422;
}

.bnLiGR:hover .sc-VigVT {
  opacity: 1;
}

<div
  className="sc-gqjmRU lhXvrk"
>
  <a
    className="sc-jTzLTM bnLiGR"
    href="Java.driver"
    target="_blank"
  >
    <img
      alt="Driver code repository on GitHub"
      className="sc-VigVT jUlwLK"
      src="github.svg"
      title="Driver code repository on GitHub"
    />
  </a>
</div>
`;

exports[`Header has the button disabled if canParse is false 1`] = `
.dfgKLf {
  height: 70px;
  padding: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  z-index: 9999;
  box-shadow: 0 5px 25px rgba(138, 128, 115, 0.14);
}

.gRdHVw {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  font-weight: normal;
  padding: 0;
  margin: 0;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.jSEybS {
  height: 40px;
}

.jaIsHC {
  margin-left: .8rem;
}

.gkhDQm {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1rem;
  height: 100%;
}

.iUXYdf {
  color: grey;
  margin-right: 1em;
  font-size: .8rem;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  color: #636262;
}

.eJfGpj {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.eJfGpj + .eJfGpj {
  margin-left: 1rem;
}

.eJfGpj:last-child {
  border-right: none;
}

.jdktxT {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-right: 0;
}

.jdktxT + .jdktxT {
  margin-left: 1rem;
}

.jdktxT:last-child {
  border-right: none;
}

.bKFJVy {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: .5em .5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: .8rem;
  font-size: .7rem;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.bKFJVy + button {
  margin-left: 3px;
}

.eMoLEz {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: .5em .5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: .8rem;
  -webkit-text-transform: none;
  text-transform: none;
}

.eMoLEz + button {
  margin-left: 3px;
}

.Sudzm {
  padding: .5em .5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: .8rem;
  -webkit-letter-spacing: .05em;
  -moz-letter-spacing: .05em;
  -ms-letter-spacing: .05em;
  letter-spacing: .05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.Sudzm:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.Sudzm + .Sudzm {
  margin-left: .7rem;
}

.khhhna {
  padding: .5em .5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: .8rem;
  -webkit-letter-spacing: .05em;
  -moz-letter-spacing: .05em;
  -ms-letter-spacing: .05em;
  letter-spacing: .05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  padding: .7rem 1.8rem;
  background: #EB7422;
  color: #FFFFFF;
  white-space: nowrap;
}

.khhhna:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.khhhna:hover {
  box-shadow: 0 5px 15px rgba(191, 76, 50, 0.48);
}

.lhXvrk {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: .5rem;
}

.jUlwLK {
  width: 20px;
  opacity: .7;
  margin-right: .3rem;
}

.bnLiGR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: .9rem;
}

.bnLiGR:hover {
  color: #EB7422;
}

.bnLiGR:hover .sc-VigVT {
  opacity: 1;
}

<header
  className="sc-bdVaJa dfgKLf"
>
  <h1
    className="sc-bwzfXH gRdHVw"
  >
    <img
      alt="bblfsh"
      className="sc-htpNat jSEybS"
      src="babelfish_logo.svg"
    />
    <span
      className="sc-bxivhb jaIsHC"
    >
      Dashboard
    </span>
  </h1>
  <div
    className="sc-ifAKCX gkhDQm"
  >
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje iUXYdf"
        htmlFor="language-selector"
      >
        Language
      </label>
      <select
        className="sc-htoDjs bKFJVy"
        id="language-selector"
        onChange={undefined}
        value="auto"
      >
        <option
          value="auto"
        >
          Python (auto)
        </option>
        <option
          value="python"
        >
          Python
        </option>
        <option
          value="java"
        >
          Java
        </option>
      </select>
      <div
        className="sc-gqjmRU lhXvrk"
      >
        <a
          className="sc-jTzLTM bnLiGR"
          href="Python.driver"
          target="_blank"
        >
          <img
            alt="Driver code repository on GitHub"
            className="sc-VigVT jUlwLK"
            src="github.svg"
            title="Driver code repository on GitHub"
          />
        </a>
      </div>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje iUXYdf"
        htmlFor="examples-selector"
      >
        Examples
      </label>
      <select
        className="sc-htoDjs bKFJVy"
        id="examples-selector"
        onChange={undefined}
        value={undefined}
      >
        <option
          value="python"
        />
        <option
          value="java"
        />
      </select>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <input
        className="sc-dnqmqq eMoLEz"
        onChange={[Function]}
        placeholder="raw gist url"
        type="url"
        value=""
      />
      <button
        className="sc-iwsKbI Sudzm"
        disabled={true}
        onClick={[Function]}
      >
        load
      </button>
      <button
        className="sc-iwsKbI Sudzm"
        disabled={true}
        onClick={[Function]}
      >
        share
      </button>
    </div>
    <div
      className="sc-gzVnrw jdktxT"
    >
      <button
        className="sc-gZMcBi khhhna"
        disabled={true}
        id="run-parser"
        onClick={undefined}
      >
        Run parser
      </button>
    </div>
  </div>
</header>
`;

exports[`Header has the button enabled if canParse is true 1`] = `
.dfgKLf {
  height: 70px;
  padding: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  z-index: 9999;
  box-shadow: 0 5px 25px rgba(138, 128, 115, 0.14);
}

.gRdHVw {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  font-weight: normal;
  padding: 0;
  margin: 0;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.jSEybS {
  height: 40px;
}

.jaIsHC {
  margin-left: .8rem;
}

.gkhDQm {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1rem;
  height: 100%;
}

.iUXYdf {
  color: grey;
  margin-right: 1em;
  font-size: .8rem;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  color: #636262;
}

.eJfGpj {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.eJfGpj + .eJfGpj {
  margin-left: 1rem;
}

.eJfGpj:last-child {
  border-right: none;
}

.jdktxT {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-right: 0;
}

.jdktxT + .jdktxT {
  margin-left: 1rem;
}

.jdktxT:last-child {
  border-right: none;
}

.bKFJVy {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: .5em .5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: .8rem;
  font-size: .7rem;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.bKFJVy + button {
  margin-left: 3px;
}

.eMoLEz {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: .5em .5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: .8rem;
  -webkit-text-transform: none;
  text-transform: none;
}

.eMoLEz + button {
  margin-left: 3px;
}

.Sudzm {
  padding: .5em .5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: .8rem;
  -webkit-letter-spacing: .05em;
  -moz-letter-spacing: .05em;
  -ms-letter-spacing: .05em;
  letter-spacing: .05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.Sudzm:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.Sudzm + .Sudzm {
  margin-left: .7rem;
}

.khhhna {
  padding: .5em .5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: .8rem;
  -webkit-letter-spacing: .05em;
  -moz-letter-spacing: .05em;
  -ms-letter-spacing: .05em;
  letter-spacing: .05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  padding: .7rem 1.8rem;
  background: #EB7422;
  color: #FFFFFF;
  white-space: nowrap;
}

.khhhna:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.khhhna:hover {
  box-shadow: 0 5px 15px rgba(191, 76, 50, 0.48);
}

.lhXvrk {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: .5rem;
}

.jUlwLK {
  width: 20px;
  opacity: .7;
  margin-right: .3rem;
}

.bnLiGR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: .9rem;
}

.bnLiGR:hover {
  color: #EB7422;
}

.bnLiGR:hover .sc-VigVT {
  opacity: 1;
}

<header
  className="sc-bdVaJa dfgKLf"
>
  <h1
    className="sc-bwzfXH gRdHVw"
  >
    <img
      alt="bblfsh"
      className="sc-htpNat jSEybS"
      src="babelfish_logo.svg"
    />
    <span
      className="sc-bxivhb jaIsHC"
    >
      Dashboard
    </span>
  </h1>
  <div
    className="sc-ifAKCX gkhDQm"
  >
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje iUXYdf"
        htmlFor="language-selector"
      >
        Language
      </label>
      <select
        className="sc-htoDjs bKFJVy"
        id="language-selector"
        onChange={undefined}
        value="auto"
      >
        <option
          value="auto"
        >
          Python (auto)
        </option>
        <option
          value="python"
        >
          Python
        </option>
        <option
          value="java"
        >
          Java
        </option>
      </select>
      <div
        className="sc-gqjmRU lhXvrk"
      >
        <a
          className="sc-jTzLTM bnLiGR"
          href="Python.driver"
          target="_blank"
        >
          <img
            alt="Driver code repository on GitHub"
            className="sc-VigVT jUlwLK"
            src="github.svg"
            title="Driver code repository on GitHub"
          />
        </a>
      </div>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje iUXYdf"
        htmlFor="examples-selector"
      >
        Examples
      </label>
      <select
        className="sc-htoDjs bKFJVy"
        id="examples-selector"
        onChange={undefined}
        value={undefined}
      >
        <option
          value="python"
        />
        <option
          value="java"
        />
      </select>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <input
        className="sc-dnqmqq eMoLEz"
        onChange={[Function]}
        placeholder="raw gist url"
        type="url"
        value=""
      />
      <button
        className="sc-iwsKbI Sudzm"
        disabled={true}
        onClick={[Function]}
      >
        load
      </button>
      <button
        className="sc-iwsKbI Sudzm"
        disabled={true}
        onClick={[Function]}
      >
        share
      </button>
    </div>
    <div
      className="sc-gzVnrw jdktxT"
    >
      <button
        className="sc-gZMcBi khhhna"
        disabled={false}
        id="run-parser"
        onClick={undefined}
      >
        Run parser
      </button>
    </div>
  </div>
</header>
`;

exports[`Header renders correctly 1`] = `
.dfgKLf {
  height: 70px;
  padding: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  z-index: 9999;
  box-shadow: 0 5px 25px rgba(138, 128, 115, 0.14);
}

.gRdHVw {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  font-weight: normal;
  padding: 0;
  margin: 0;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.jSEybS {
  height: 40px;
}

.jaIsHC {
  margin-left: .8rem;
}

.gkhDQm {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1rem;
  height: 100%;
}

.iUXYdf {
  color: grey;
  margin-right: 1em;
  font-size: .8rem;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  color: #636262;
}

.eJfGpj {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.eJfGpj + .eJfGpj {
  margin-left: 1rem;
}

.eJfGpj:last-child {
  border-right: none;
}

.jdktxT {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-right: 0;
}

.jdktxT + .jdktxT {
  margin-left: 1rem;
}

.jdktxT:last-child {
  border-right: none;
}

.bKFJVy {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: .5em .5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: .8rem;
  font-size: .7rem;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.bKFJVy + button {
  margin-left: 3px;
}

.eMoLEz {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: .5em .5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: .8rem;
  -webkit-text-transform: none;
  text-transform: none;
}

.eMoLEz + button {
  margin-left: 3px;
}

.Sudzm {
  padding: .5em .5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: .8rem;
  -webkit-letter-spacing: .05em;
  -moz-letter-spacing: .05em;
  -ms-letter-spacing: .05em;
  letter-spacing: .05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.Sudzm:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.Sudzm + .Sudzm {
  margin-left: .7rem;
}

.khhhna {
  padding: .5em .5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: .8rem;
  -webkit-letter-spacing: .05em;
  -moz-letter-spacing: .05em;
  -ms-letter-spacing: .05em;
  letter-spacing: .05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  padding: .7rem 1.8rem;
  background: #EB7422;
  color: #FFFFFF;
  white-space: nowrap;
}

.khhhna:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.khhhna:hover {
  box-shadow: 0 5px 15px rgba(191, 76, 50, 0.48);
}

.lhXvrk {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: .5rem;
}

.jUlwLK {
  width: 20px;
  opacity: .7;
  margin-right: .3rem;
}

.bnLiGR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: .9rem;
}

.bnLiGR:hover {
  color: #EB7422;
}

.bnLiGR:hover .sc-VigVT {
  opacity: 1;
}

<header
  className="sc-bdVaJa dfgKLf"
>
  <h1
    className="sc-bwzfXH gRdHVw"
  >
    <img
      alt="bblfsh"
      className="sc-htpNat jSEybS"
      src="babelfish_logo.svg"
    />
    <span
      className="sc-bxivhb jaIsHC"
    >
      Dashboard
    </span>
  </h1>
  <div
    className="sc-ifAKCX gkhDQm"
  >
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje iUXYdf"
        htmlFor="language-selector"
      >
        Language
      </label>
      <select
        className="sc-htoDjs bKFJVy"
        id="language-selector"
        onChange={undefined}
        value="auto"
      >
        <option
          value="auto"
        >
          Java (auto)
        </option>
        <option
          value="python"
        >
          Python
        </option>
        <option
          value="java"
        >
          Java
        </option>
      </select>
      <div
        className="sc-gqjmRU lhXvrk"
      >
        <a
          className="sc-jTzLTM bnLiGR"
          href="Java.driver"
          target="_blank"
        >
          <img
            alt="Driver code repository on GitHub"
            className="sc-VigVT jUlwLK"
            src="github.svg"
            title="Driver code repository on GitHub"
          />
        </a>
      </div>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje iUXYdf"
        htmlFor="examples-selector"
      >
        Examples
      </label>
      <select
        className="sc-htoDjs bKFJVy"
        id="examples-selector"
        onChange={undefined}
        value={undefined}
      >
        <option
          value="python"
        />
        <option
          value="java"
        />
      </select>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <input
        className="sc-dnqmqq eMoLEz"
        onChange={[Function]}
        placeholder="raw gist url"
        type="url"
        value=""
      />
      <button
        className="sc-iwsKbI Sudzm"
        disabled={true}
        onClick={[Function]}
      >
        load
      </button>
      <button
        className="sc-iwsKbI Sudzm"
        disabled={true}
        onClick={[Function]}
      >
        share
      </button>
    </div>
    <div
      className="sc-gzVnrw jdktxT"
    >
      <button
        className="sc-gZMcBi khhhna"
        disabled={true}
        id="run-parser"
        onClick={undefined}
      >
        Run parser
      </button>
    </div>
  </div>
</header>
`;
